brainstorming.md

```diff
@@ -0,0 +1,508 @@
+# Introduction
+
+> **Status**: brainstorming, pre-proposal
+>
+> ℹ️ **Note** - this is an early draft and has not had any substantive review or circulation. It may contain controversial, undecided, or just plain wrong-headed ideas. I expect several major iterations will be required to converge the underlying primitives with use cases. Please see `issues` noted throughout the doc for a list of active discussions/debates. In particular, this doc has many known gaps identified throughout the with callouts.
+
+The goal of SOMA (“stack of matrices, annotated”) is a flexible, extensible, and open-source API providing access to annotated, 2D matrix data stored in multiple underlying formats and systems. The vision for this API family includes:
+
+- support access to persistent, cloud-resident datasets,
+- enable use within popular data science environments (eg, R, Python), using the tools of that environment (eg, Python Pandas integration),
```

Comment:
i'll leave word-smithing large & small (example of the latter: _eg_ -> _e.g._) until separate review

Reply:
keep it up and you will be assigned editorial responsibilities :-(

Reply:
@bkmartinjr tbh I enjoy it! :) Used to work as a copy-editor for a brief while, & was happy at it :)

Reply:
you have a calling!

For now, lets focus on those issues that are misleading, block comprehension, etc.  There will definitely be a cleanup and "add examples" pass at some point, after we agree on the proposal.  I will be _very_ happy to have assistance at that point, as I am (demonstrably) an amateur at this "writing thing".

Reply:
"eg" and "ie" are fixed.